features/pages/flaw_detail_page.py

```diff
@@ -334,39 +346,63 @@ def add_reference_select_external_type(self):
         except ElementNotVisibleException:
             pass
 
-        self.addReferenceSelect.select_element_by_text("External")
+        add_reference_select = self.driver.find_elements(
+            locate_with(By.TAG_NAME, "select").
+            below(self.referenceCountLabel))[0]
+
+        select = Select(add_reference_select)
+        select.select_by_visible_text("External")
 
     def add_reference_set_link_url(self, value):
-        self.driver.execute_script("arguments[0].scrollIntoView(true);", self.addReferenceLinkUrlInput)
-        self.addReferenceLinkUrlInput.set_text(value)
+        add_reference_link_url_input = self.driver.find_elements(
+            locate_with(By.TAG_NAME, "input").
+            near(self.addReferenceLinkUrlSpan))[0]
```

Comment:
Scenario: Modify reference                        # features/flaw_detail.feature:89
    Given I am an analyst AND I am logged into OSIM # features/steps/common_steps.py:13 8.978s
    And I set the bugzilla api key and jira api key # features/steps/common_steps.py:18 5.521s
    Given I go to a public flaw detail page         # features/steps/common_steps.py:29 1.746s
    When I edit a internal/external reference       # features/steps/flaw_detail.py:370 15.740s
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/seleniumpagefactory/Pagefactory.py", line 53, in __getattr__
          element = WebDriverWait(self.driver, self.timeout).until(
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/support/wait.py", line 101, in until
          raise TimeoutException(message, screen, stacktrace)
      selenium.common.exceptions.TimeoutException: Message: 
      Stacktrace:
      RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
      WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5
      NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:509:5
      dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16
      
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/flaw_detail.py", line 385, in step_impl
          flaw_detail_page.add_reference_set_link_url(context.expected)
        File "/Users/huiwang/issue-2640/osim/features/pages/flaw_detail_page.py", line 359, in add_reference_set_link_url
          near(self.addReferenceLinkUrlSpan))[0]
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/seleniumpagefactory/Pagefactory.py", line 57, in __getattr__
          raise ElementNotFoundException(
      seleniumpagefactory.Pagefactory.ElementNotFoundException: An exception of type TimeoutException occurred. With Element -: addReferenceLinkUrlSpan - locator: (xpath, //span[text()='Link URL'])

Reply:
Fixed